zebra/zebra_rib.c

```diff
@@ -1414,6 +1416,22 @@ static void rib_process(struct route_node *rn)
 		}
 	}
 
+	/*
+	 * If zebra has a new_selected and a proto_re_changed
+	 * entry that was not the old selected and the protocol
+	 * is different, zebra should notify the upper level
+	 * protocol that the sent down entry was not selected
+	 */
+	if (new_selected && proto_re_changed &&
```

Comment:
if we're going to keep this, could you add a line to the comment block above, the block that begins "After the cycle is finished..." - that describes the variables that are going to be checked after the loop through the REs, and it'd be helpful to include the new variable there too.